src/Compilers/Core/Portable/Emit/EditAndContinue/EncVariableSlotAllocator.cs

```diff
@@ -319,5 +328,21 @@ public override bool TryGetPreviousLambda(SyntaxNode lambdaOrLambdaBodySyntax, b
             lambdaId = default;
             return false;
         }
+
+        public override int? GetFirstUnusedStateMachineState(bool increasing)
+            => increasing ? _firstUnusedIncreasingStateMachineState : _firstUnusedDecreasingStateMachineState;
+
+        public override bool TryGetPreviousStateMachineState(SyntaxNode syntax, out int stateOrdinal)
```

Comment:
nit: consider asserting the syntax is expected kind as described in abstract API: "<paramref name="syntax"/> is an await expression, yield return statement, or try block syntax node.". That would probably need to use hard-coded/known integers for the respective C# and VB syntax kinds.

I didn't quite follow why the syntax could be a try block here. I thought we care to map suspension points only.

Reply:
> nit: consider asserting the syntax is expected kind as described in abstract API: " is an await expression, yield return statement, or try block syntax node.". That would probably need to use hard-coded/known integers for the respective C# and VB syntax kinds.

Since this is language-agnostic code it'd need to compare raw kind to integers. I put the asserts to the language-specific call sites instead.